store/precomputed_key/wvm/rpc/wvm_rpc_client.go

```diff
@@ -0,0 +1,241 @@
+package wvm
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	wvmtypes "github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/wvm/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type Signer interface {
+	GetAccount(ctx context.Context) (common.Address, error)
+	SignTransaction(ctx context.Context, signData *wvmtypes.SignData) (string, error)
+}
+
+// WVM RPC client
+type RPCClient struct {
+	log     log.Logger
+	client  *ethclient.Client
+	chainID int64
+	signer  Signer
+}
+
+func NewWvmRPCClient(log log.Logger, cfg *wvmtypes.Config, signer Signer) (*RPCClient, error) {
```

Comment:
is this shadowing an existing client implementation or is it written from scratch?

Reply:
mmm,
to make standard calls this package utilizes go-ethereum/ethclient and types 
cause our node is a fork of ethereum client(in this case - reth) we are free from rewriting this part from scratch

everything else written around it